sdk/src/builder.rs

```diff
@@ -1239,6 +1250,7 @@ impl Builder {
         ingredient_map: &HashMap<String, (&Relationship, HashedUri)>,
         actions: &mut Actions,
         settings: &Settings,
+        intent: Option<BuilderIntent>,
     ) -> Result<()> {
```

Comment:
I think it would make more sense if this function took `&self` and used `self.settings` and `self.intent`.